### PR TITLE
Enable using PKCS#12 based Java keystores

### PR DESCRIPTION
**Pull request by modulo11**
## Summary

- Part of https://github.com/paketo-buildpacks/libjvm/issues/158
- Abstracts operations on Java keystores by introducing an interface for common operations
- Minimum API, closely alight to previous implementation

## Use Cases

- Re-enables adding ca-certificates to the Java keystore for Java 18 and above

## Checklist

* [X] I have viewed, signed, and submitted the Contributor License Agreement.
* [X] I have linked issue(s) that this PR should close using keywords or the Github UI (See [docs](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue))
* [X] I have added an integration test, if necessary.
* [X] I have reviewed the [styleguide](https://github.com/paketo-buildpacks/community/blob/main/STYLEGUIDE.md) for guidance on my code quality.
* [X] I'm happy with the commit history on this PR (I have rebased/squashed as needed).

### CODE REVIEW

File: keystore.go [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+	location string
+	entries  []pkcs12.TrustStoreEntry
+}
+
+func NewPKCS12Keystore(location string) *PKCS12Keystore {
+	return &PKCS12Keystore{
+		location: location,
+		entries:  []pkcs12.TrustStoreEntry{},
+	}
+}
```
Comment (dmikusa) [inline keystore.go:127]: So this is for a password-less PKCS12 keystore, right?

If so...

1. Do you think we need to support passwords? I think all of the JVM vendors are using passwordless with their defaults, so from that perspective, it seems unnecessary. Unless you have a different JVM vendor, or someone's customized the default keystore 🤔 
2. If we don't support this now and if we were to need to support it at some point in the future, do you think we could add that to the existing code in a non-breaking way? If not, do you think we should rename this implementation PasswordLessPKCS12Keystore or something like that to leave room for a new implementation that does support passwords?

I don't think this is a huge issue, but just trying to be forward-thinking about this.

**Review by dmikusa — COMMENTED** [review]
Code looks great! Thanks for putting this together. Just one question for you. 

In the meantime, I'm going to give this a try with a few buildpacks and see how it runs.

**Comment by dmikusa** [comment]
Tested with Bellsoft buildpack:
 - current version with Java 21 displays warning and does not load certs at runtime
 - new version of libjvm same buildpack version with Java 21, loads certs at build time and runtime (no warning)
 - new version of libjvm same buildpack version with Java 17, works as it did before
 - looks correct

Tested with Adoptium, same behavior. Working correctly.

Tested with Azul Zulu, same behavior. Working correctly.

I'm happy with the testing.

@modulo11 did you test with the sap-machine buildpack? 
@anthonydahanne or @pivotal-david-osullivan do you want to test this as well? or good to merge?

File: keystore.go (hunk 1 above)
Comment (modulo11) [reply keystore.go:127]: I don't think that we need to support passwords, but I can check internally if there's some guidance. Renaming the implementation, to reflect that it does not use a password, seems reasonable to me. That way, we can introduce the functionality later without breaking anyone.

**Comment by modulo11** [comment]
> @modulo11 did you test with the sap-machine buildpack? @anthonydahanne or @pivotal-david-osullivan do you want to test this as well? or good to merge?

Yes, tested locally with SapMachine and seems to work nicely.

**Review by dmikusa — APPROVED** [review]

**Review by anthonydahanne — APPROVED** [review]

**Pull request merged**